File: src/MicroServicioAuth/Entities/LoginResponse.cs
namespace MicroServicioAuth.Entities
{
    /// <summary>
    /// Cuerpo de la respuesta exitosa (HTTP 201) del endpoint <c>POST /auth/login</c>.
    /// Contiene los tokens de acceso y la información de sesión del usuario autenticado.
    /// </summary>
    public class LoginResponse
    {
        /// <summary>Fecha y hora UTC en que vence el JWT de acceso.</summary>
        public DateTime expires_in { get; set; }

        /// <summary>JWT firmado con HMAC-SHA256 para autenticar solicitudes.</summary>
        public string access_token { get; set; } = string.Empty;

        /// <summary>Token de refresco para renovar la sesión cuando el JWT expire.</summary>
        public string refresh_token { get; set; } = string.Empty;

        /// <summary>Email del usuario autenticado (identificador de sesión).</summary>
        public string usuarioID { get; set; } = string.Empty;

        /// <summary>RolID de la fila correspondiente en UsuariosXInstituciones.</summary>
        public Guid rol { get; set; }
    }
}

File: src/app_usr/lib/models/login_response.dart
/// Respuesta del POST /gateway/auth/login
class LoginResponse {
  final String accessToken;
  final String refreshToken;
  final String expiresIn;
  // El campo usuarioID en la respuesta del microservicio corresponde al email
  final String usuarioId;

  LoginResponse({
    required this.accessToken,
    required this.refreshToken,
    required this.expiresIn,
    required this.usuarioId,
  });

  factory LoginResponse.fromJson(Map<String, dynamic> json) {
    return LoginResponse(
      accessToken:  json['access_token']  as String,
      refreshToken: json['refresh_token'] as String,
      expiresIn:    json['expires_in'].toString(),
      usuarioId:    json['usuarioID']     as String,
    );
  }
}